test/util/test_path_provider.py

```diff
@@ -0,0 +1,173 @@
+import os
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+
+class TestPathProviderBase(ABC):
+    """
+        used to generate the files path's in all tests
+        in particular this file is helping with the integration between running test_integration script
+        and the integration_test_runner.
+        all test path's should go through here.
+        in the future should use the constants module to be more stable.
+        we use abstract class so that every enviorment could easily setup it's own files structure.
```

Comment:
this is an abstract class? what makes it so?

Reply:
note it implements abs and uses the abstractmethod decorator over abstract methods.